packages/protocol-kit/src/contracts/utils.ts

```diff
@@ -240,3 +236,37 @@ export const validateSafeDeploymentConfig = ({ saltNonce }: SafeDeploymentConfig
   if (saltNonce && BigInt(saltNonce) < 0)
     throw new Error('saltNonce must be greater than or equal to 0')
 }
+
+/**
+ * Generates a zkSync Era address. zkSync Era uses a distinct address derivation method compared to Ethereum
+ *
+ * @param {string} from - The sender's address.
+ * @param {SafeVersion} safeVersion - The version of the safe.
+ * @param {Buffer} salt - The salt used for address derivation.
+ * @param {string} input - Additional input data for the derivation.
+ *
+ * @returns {string} The derived zkSync Era address.
+ */
+export function zkSyncEraCreate2Address(
```

Comment:
Could you link the source code that inspired this implementation? I think will be helpful in the future

Reply:
sure!